src/System.IO.Pipelines.Networking.Libuv/UvTcpConnection.cs

```diff
@@ -214,6 +223,8 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
                         }
                     }
                 }
+
+                inputBufferHandle.Free();
```

Comment:
> Hmm sorry, do you mind explaining why that's the case? 

Why would that be a problem? The pin only needs to last as long as the native call that copies data into it. From then the GC is free to move it around. Once the OnRead callback is called it can be freed, otherwise, you may as well keep the `TryGetPointer` calls.

> If we commit but keep the buffer then wouldn't it be a problem to drop the handle if it's still going to be used?

Why? `OnAlloc` would be called again and we would pin the buffer again. The cases where `OnRead` gets called before `OnAlloc` are pretty rare. The usual pattern is `OnAlloc`, `OnRead`